test/e2e/framework/resource_usage_gatherer.go

```diff
@@ -237,13 +237,20 @@ func NewResourceUsageGatherer(c clientset.Interface, options ResourceGathererOpt
 				return nil, err
 			}
 		}
+		dnsNodes := make(map[string]bool)
 		for _, pod := range pods.Items {
+			if options.MasterOnly && !system.IsMasterNode(pod.Spec.NodeName) && pod.Labels["k8s-app"] != "kube-dns" {
+				continue
+			}
 			for _, container := range pod.Status.InitContainerStatuses {
 				g.containerIDs = append(g.containerIDs, container.Name)
 			}
 			for _, container := range pod.Status.ContainerStatuses {
 				g.containerIDs = append(g.containerIDs, container.Name)
 			}
+			if options.MasterOnly {
+				dnsNodes[pod.Spec.NodeName] = true
```

Comment:
I'm not a fan of this.
DNS pods are not running on master VM. So hacking them into "MasterOnly" seems wrong to me.

Instead the MasterOnly option should be changed to some enum, with values:
MasterOnly
MasterAndDNS
Everything
[better names appreciated]

Reply:
OK renamed to enum ...  

AllNodes
MasterNodes
MasterAndDNSNodes